lib/Workflow/Condition.pm

```diff
@@ -61,14 +62,51 @@ sub evaluate_condition {
         $condition = $wf->_factory()
             ->get_condition( $orig_condition, $wf->type );
         $log->debug( "Evaluating condition '$orig_condition'" );
-        my $return_value = $condition->evaluate($wf);
+
+        my $return_value;
+        my $result = $condition->evaluate($wf);
+        if (ref $result eq 'Workflow::Condition::IsTrue') {
+            $log->info( "Got true result with '$result' on '$orig_condition'");
```

Comment:
There's one (big?) downside to this approach (using centralized logging): if you want to do selective logging (like the class-based logging that Log4perl allows), then it won't work for condition checks.

Reply:
You can also add logging into the conditions classes so I dont see this as a point - in this place I am only interested on the result of the named condition and do not care on the inner details of this decision. If you can come up with a "better" solution I am happy to discuss this but we have quite the same logger in the old code :)

Reply:
Ok. So, I've pushed a change which lets you select between strict boolean (class-based) and regular values.